contrib/views/capacity-scheduler/src/main/java/org/apache/ambari/view/capacityscheduler/ConfigurationService.java

```diff
@@ -316,7 +316,7 @@ public Response getRmSchedulerConfig() {
    */
   private boolean isOperator() {
 
-    String url = String.format(AMBARI_OR_CLUSTER_ADMIN_PRIVILEGE_URL, context.getUsername(), context.getCluster().getName());
+    String url = String.format(AMBARI_OR_CLUSTER_ADMIN_PRIVILEGE_URL, context.getLoggedinUser(), context.getCluster().getName());
```

Comment:
Why was this change needed.  For example, what was the value of `context.getUsername()` that was incorrect and what is the value of `context.getLoggedinUser()` that is correct?

Reply:
Hi @rlevas ,

Refering to : https://github.com/apache/ambari/blob/5460e8952729854f1c032a781c9a8de608ba4475/ambari-server/src/main/java/org/apache/ambari/server/view/ViewContextImpl.java#L233

The context.getUsername() will return a authToLocal convered username and context. getLoggedinUser() will return the current ambari username, the Yarn CapacityScheduler needs only current ambari username as its used by it to validate whether user is authorized to edit the ambari config or not.


So i assume context.getLoggedInUser() is enough..

the problem here is yarn caapcityScheduler is failing to authorize if some authToLocal rules are set and ambari username is satisfying the same.

Reply:
Thanks for the clarification.